pydevices/acq400Devices/acq2106_mgtdram8.py

```diff
@@ -0,0 +1,327 @@
+                    # timeout exception is setup
+
+                    if err == 'timed out':
+                        time.sleep(1)
+                        print (' recv timed out, retry later')
+                        # continue                  else:
+                        print (e)
+                        # break
+                except socket.error as e:
+                    # Something else happened, handle error, exit, etc.
+                    print("socket error", e)
+                    self.full_buffers.put(buf[:self.segment_bytes-toread])
+
+
+    def setChanScale(self,num):
+        chan=self.__getattr__(INPFMT % num)
+        chan.setSegmentScale(MDSplus.ADD(MDSplus.MULTIPLY(chan.COEFFICIENT,MDSplus.dVALUE()),chan.OFFSET))
+
+    def lazy_init(self):
+        try:
+            return self.uut
+        except AttributeError:
+            self.uut = acq400_hapi.Acq2106_Mgtdram8(self.node.data(), monitor=False)
+            return self.uut        
+            
+    def init(self):
+        print('Running init')
+        self.lazy_init()
+
+        trg = self.trig_mode.data()
+
+        # The default case is to use the trigger set by sync_role.
+        if self.trig_mode.data() == 'role_default':
+            self.uut.s0.sync_role = "{} {}".format(self.role.data(), self.freq.data())
+        else:
+            # If the user has specified a trigger.
+            self.uut.s0.sync_role = '{} {} TRG:DX={}'.format(self.role.data(), self.freq.data(), trig_types[trg])
+
+        # Now we set the trigger to be soft when desired.
+        if trg == 'soft':
+            self.uut.s0.transient = 'SOFT_TRIGGER=0'
+        if trg == 'automatic':
+            self.uut.s0.transient = 'SOFT_TRIGGER=1'
+
+        print('Finished init')
+
+    def capture(self):
+        print("Capturing now.")
+        self.lazy_init()
+        self.uut.s14.mgt_taskset = '1'
+        self.uut.s14.mgt_run_shot = str(self.cap_blocks + 2)
+        self.uut.run_mgt()
+        print("Finished capture.")
+    CAPTURE=capture
+
+    def soft_trigger(self): 
+        self.lazy_init()
+        self.uut.s0.soft_trigger = 1
+
+
+    def pull(self):
+        print("Starting host pull now.")
+        self.lazy_init()
+        
+        import os
+        import sys
+        from threading import Thread
+        import tempfile
+        import subprocess
+
+        self.uut.s0.set_knob('set_abort', '1')
+        self.running.on=True        
+        thread = self.MDSWorker(self, self.uut.nchan())
+        thread.start()
+    PULL=pull
+
+
+    def stop(self):
+        self.running.on = False
+    STOP=stop
+
+
+    
+def assemble(cls):     
+    cls.parts = list(_ACQ2106_MGTDRAM8.base_parts)
+    for ch in range(1, cls.nchan+1):
+        cls.parts.append({'path':INPFMT%(ch,), 'type':'signal','options':('no_write_model','write_once',),
+                          'valueExpr':'head.setChanScale(%d)' %(ch,)})
```

Comment:
unless something has changed but does this not depend on the gain setting of the channel and should be read per shot